jira_access/_jira_client.py

```diff
@@ -0,0 +1,105 @@
+                                  api_action: str,
+                                  params: Optional[Mapping[str, str]] = None,
+                                  headers: Optional[Mapping[str, str]] = None) -> Json | JsonType:
+        headers = headers or self._DEFAULT_HEADERS
+        url = self._api_url + api_action
+        r: requests.Response = requests.get(url,
+                                            params=params,
+                                            headers=headers,
+                                            auth=self._auth,
+                                            timeout=self._REQUEST_TIMEOUT_SEC)
+
+        if not r.ok:
+            if r.status_code == 400:
+                raise Exception(f"Error executing query '{params}'.\nError message: {r.json()}\nURL: {r.url}")
+            r.raise_for_status()
+        return r.json()
+
+    def _execute_http_post_requests_from_json_file(self,
+                                                   json_file_path: str,
+                                                   api_action: str,
+                                                   params: Optional[Mapping[str, str]] = None,
+                                                   headers: Optional[Mapping[str, str]] = None) -> Json | JsonType:
+        with open(json_file_path, "rb") as data:
+            return self._execute_http_post_request(params=params, data=data, api_action=api_action, headers=headers)
+
+    def _execute_http_post_request(self,
+                                   data: Json | IO,
+                                   api_action: str,
+                                   params: Optional[Mapping[str, str]] = None,
+                                   headers: Optional[Mapping[str, str]] = None) -> Json | JsonType:
+        headers = headers or self._DEFAULT_HEADERS
+        payload = json.dumps(data) if isinstance(data, dict) else data
+        url = self._api_url + api_action
+        r: requests.Response = requests.post(url,
+                                             params=params,
+                                             data=payload,
+                                             headers=headers,
+                                             auth=self._auth,
+                                             timeout=self._REQUEST_TIMEOUT_SEC)
+
+        if not r.ok:
+            if r.status_code == 400:
+                raise Exception(f"Error executing put request.\nError message: {r.json()}\nURL: {r.url}")
+            r.raise_for_status()
+        return r.json()
+
+    def _execute_http_put_requests_from_json_file(self,
+                                                  json_file_path: str,
+                                                  api_action: str,
+                                                  params: Optional[Mapping[str, str]] = None,
+                                                  headers: Optional[Mapping[str, str]] = None) -> Json | JsonType:
```

Comment:
Why this shall return either a Json or JsonType ?

Reply:
You are right. It should be simply JsonType.